apps/extension/src/routes/popup/settings/settings-connected-sites.tsx

```diff
@@ -0,0 +1,122 @@
+import { Link1Icon, LinkBreak1Icon, MagnifyingGlassIcon, TrashIcon } from '@radix-ui/react-icons';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { Button, FadeTransition, Input } from '@penumbra-zone/ui';
+import { LinkGradientIcon } from '../../../icons';
+import { SettingsHeader } from '../../../shared';
+import { OriginRecord, localExtStorage } from '@penumbra-zone/storage';
+import { DisplayOriginURL } from '../../../shared/components/display-origin-url';
+import Map from '@penumbra-zone/polyfills/Map.groupBy';
+
+export const SettingsConnectedSites = () => {
+  const [search, setSearch] = useState<string>();
+
+  const [sitesFromStorage, setSitesFromStorage] = useState<OriginRecord[]>();
+
+  const loadSitesFromStorage = useCallback(
+    async () => setSitesFromStorage(await localExtStorage.get('connectedSites')),
+    [],
+  );
+  useEffect(() => void loadSitesFromStorage(), [loadSitesFromStorage]);
```

Comment:
This is supposed to run on mount right? This pattern is a bit non-standard. Think you can do this:

```typescript
  useEffect(() => {
    void (async () => {
      const sites = await localExtStorage.get('connectedSites');
      setSitesFromStorage(sites);
    })();
  }, []);
```